aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/Merger.java

```diff
@@ -77,22 +76,24 @@
             // Update the metadata for it
             .andThen(save(metadata))
             // Let the world know that we've done a good thing.
-            .andThen(announceSuccessfulMerge(modelWithMetadata));
+            .doOnComplete(() -> {
+                announceSuccessfulMerge(modelWithMetadata);
```

Comment:
Since we're doing two things now, is it worth just inlining? Or is this used elsewhere? If it is, any downside to hoisting the log statement out of it?

Reply:
If I understand your meaning -- do you mean:
```
            .doOnComplete(() -> {
                announceSuccessfulMerge(modelWithMetadata);
                LOG.verbose("Remote model update was sync'd down into local storage: " + modelWithMetadata);
            })
            .doOnError(failure ->
                LOG.warn("Failed to sync remote model into local storage: " + modelWithMetadata)
            );
```
vs.
```
            .doOnComplete(() -> announceSuccessfulMerge(modelWithMetadata))
            .doOnError(failure -> LOG.warn("Failed to sync remote model into local storage: " + modelWithMetadata));
```
With the first log moved into `announceSuccessfulMerge`?

The `announceSuccessfulMerge` published a Hub event, which is a unique path. But I wanted to keep the logging for success/failure close to each other to keep the either-or relationship clear, and nearby in the code.